src/s2wasm.h

```diff
@@ -370,22 +378,59 @@ class S2WasmBuilder {
     s = inputStart;
     while (*s) {
       skipWhitespace();
-      s = strstr(s, ".type");
-      if (!s) break;
-      mustMatch(".type");
-      Name name = getCommaSeparated();
-      skipComma();
-      if (!match("@function")) continue;
-      if (match(".hidden")) mustMatch(name.str);
-      mustMatch(name.str);
-      if (match(":")) {
-        info->implementedFunctions.insert(name);
-      } else if (match("=")) {
-        Name alias = getAtSeparated();
-        mustMatch("@FUNCTION");
-        info->aliasedFunctions.insert({name, alias});
+
+      // add function definitions and aliases
+      if (match(".type")) {
+        Name name = getCommaSeparated();
+        skipComma();
+        if (!match("@function")) continue;
+        if (match(".hidden")) mustMatch(name.str);
+        mustMatch(name.str);
+        if (match(":")) {
+          info->implementedFunctions.insert(name);
+        } else if (match("=")) {
+          Name alias = getAtSeparated();
+          mustMatch("@FUNCTION");
+          info->aliasedSymbols.insert({name, LinkerObject::SymbolAlias(alias, LinkerObject::Relocation::kFunction, 0, 0)});
+        } else {
+          abort_on("unknown directive");
+        }
+      // add data aliases
       } else {
-        abort_on("unknown directive");
+        Name lhs = getStrToSep();
```

Comment:
Do you know much about what the expectations will be for symbol aliases, e.g.
1) do we expect to define them more than once in an asm file (e.g. is your code on line 424 necessary?)
2) do we expect that the aliasee will be declared before the alias (e.g. can we assert that e.g. the sizes match?)